src/IntentBuilder.ts

```diff
@@ -209,32 +260,36 @@ export class IntentBuilder {
    * @returns The signature as a string.
    */
   private async sign(chainId: number, account: Account, builder: UserOperationBuilder) {
-    const userOp = builder.getOp();
-    const packedData = ethers.AbiCoder.defaultAbiCoder().encode(
-      ['address', 'uint256', 'bytes32', 'bytes32', 'uint256', 'uint256', 'uint256', 'uint256', 'uint256', 'bytes32'],
-      [
-        userOp.sender,
-        userOp.nonce.toString(),
-        ethers.keccak256(userOp.initCode as BytesLike),
-        ethers.keccak256(userOp.callData as BytesLike),
-        userOp.callGasLimit.toString(),
-        userOp.verificationGasLimit.toString(),
-        userOp.preVerificationGas.toString(),
-        userOp.maxFeePerGas.toString(),
-        userOp.maxPriorityFeePerGas.toString(),
-        ethers.keccak256(userOp.paymasterAndData as BytesLike),
-      ],
-    );
-
-    const enc = ethers.AbiCoder.defaultAbiCoder().encode(
-      ['bytes32', 'address', 'uint256'],
-      [ethers.keccak256(packedData), ENTRY_POINT, chainId],
-    );
-
-    const userOpHash = ethers.keccak256(enc);
+    const userOpHash = computeUserOpHash(chainId, builder);
     return await account.signer.signMessage(ethers.getBytes(userOpHash));
   }
 
+  /**
+   * xSign: Signs multiple UserOperations across different chains (cross-chain) using UserOperationBuilder.
+   * @param chainIds Array of chain IDs for cross-chain signing.
+   * @param account Account used for signing the UserOperations.
+   * @param builders UserOperationBuilder instances representing the UserOperations.
+   * @returns A promise that resolves to the signature string.
+   */
+  private async xSign(chainIds: number[], account: Account, builders: UserOperationBuilder[]) {
```

Comment:
sign and xSign sharing the same flow:
1. hash
2. sign 
3. verify (for xsign only)
I think it makes sense to merge single and multi  together

Reply:
fixed